### DISCUSSION TITLE
SRLinux support for NAT

### DISCUSSION THREAD

**Discussion by vinoroidski** [srl-labs/containerlab#752] [Q&A] [answered]
We are a lab group trying to learn how to use containerlab specifically with srlinux and are investigating how to set up NAT.

We have installed iptables but are stuck because our interfaces are not exposed (on the underlying linux). At least they are not showing up when we use the ifconfig command.

We can post our setup file and some more details if needed.
Does srlinux support NAT out-of-the-box?
Do you have any tips like tutorials or other resources we can check?

**Comment by hellt** [comment, marked as answer]
Hi @vinoroidski 
what are you trying to achieve? Can you describe the use case?

**Reply by vinoroidski** [reply]
Hi @hellt ,

first off - thanks so much for taking the time and sorry for the multiple messages :( We were not sure if we would actually reach anyone.

We are trying to emulate a Home Gateway (basically a home router) which is connected to the home network with ethernet (see topology file: https://code.fbi.h-da.de/stjolepp/acn-homegateway/-/blob/main/acn_hgw.yml and we have attached a copy here). 

[acn_hgw.zip](https://github.com/srl-labs/containerlab/files/7950008/acn_hgw.zip)

We are trying to implement functionalities like support for multiple devices within the home network, NAT, DNS, DHCP and Firewall functionality.

However we are stuck at implementing NAT with SR Linux because we could't find a way to configure it natively and as mentioned above could not get it to work by using the underlying linux.

Is there a way to get NAT work in our use case with SR Linux? We would appreciate any learning resources you could recommend (examples, manuals or tutorials) because we have searched a few weeks now and not found anything useful.

**Reply by hellt** [reply]
Hi @vinoroidski it is clearer now
so if you want SR Linux to perform a traditional NAT function - which is to translate address for packates coming in/out of a certain interface - this is not something that SR Linux supports. @brwallis may correct me on this.

As SR Linux current focus is on datacenter networks, this kind of functionality is not relevant to the domain, hence it is not there.

**Reply by vinoroidski** [reply]
I understand.  Thanks so much for the explanation and support @hellt!

We took some steps back and started our setup with Cumulus and have established a first working setup :)
